frontend/packages/console-app/src/components/nodes/NodesPage.tsx

```diff
@@ -242,162 +243,153 @@ type NodesRowMapFromStateProps = {
-            className={nodeColumnInfo.role.classes}
-            columns={columns}
-            columnID={nodeColumnInfo.role.id}
-          >
-            <NodeRoles node={node} />
-          </TableData>
-          <TableData
-            className={nodeColumnInfo.pods.classes}
-            columns={columns}
-            columnID={nodeColumnInfo.pods.id}
-          >
-            {pods}
-          </TableData>
-          <TableData
-            className={nodeColumnInfo.memory.classes}
-            columns={columns}
-            columnID={nodeColumnInfo.memory.id}
-          >
-            {memory}
-          </TableData>
-          <TableData
-            className={nodeColumnInfo.cpu.classes}
-            columns={columns}
-            columnID={nodeColumnInfo.cpu.id}
-          >
-            {cpu}
-          </TableData>
-          <TableData
-            className={nodeColumnInfo.filesystem.classes}
-            columns={columns}
-            columnID={nodeColumnInfo.filesystem.id}
-          >
-            {storage}
-          </TableData>
-          <TableData
-            className={nodeColumnInfo.created.classes}
-            columns={columns}
-            columnID={nodeColumnInfo.created.id}
-          >
-            <Timestamp timestamp={node.metadata.creationTimestamp} />
-          </TableData>
-          <TableData
-            className={nodeColumnInfo.instanceType.classes}
-            columns={columns}
-            columnID={nodeColumnInfo.instanceType.id}
-          >
-            {instanceType || '-'}
-          </TableData>
-          <TableData
-            className={nodeColumnInfo.machine.classes}
-            columns={columns}
-            columnID={nodeColumnInfo.machine.id}
-          >
-            {machine ? (
-              <ResourceLink
-                kind={referenceForModel(MachineModel)}
-                name={machine.name}
-                namespace={machine.namespace}
-              />
-            ) : (
-              '-'
-            )}
-          </TableData>
-          <TableData
-            className={nodeColumnInfo.labels.classes}
-            columns={columns}
-            columnID={nodeColumnInfo.labels.id}
-          >
-            <LabelList kind={kind} labels={labels} />
-          </TableData>
-          <TableData
-            className={nodeColumnInfo.zone.classes}
-            columns={columns}
-            columnID={nodeColumnInfo.zone.id}
-          >
-            {zone}
-          </TableData>
-          <TableData className={Kebab.columnClass}>
-            <ResourceKebab
-              actions={menuActions}
-              kind={referenceForModel(NodeModel)}
-              resource={node}
+  ({
+    obj: node,
+    index,
+    rowKey,
+    style,
+    metrics,
+  }: NodesTableRowProps & NodesRowMapFromStateProps) => {
+    const [tableColumns, , loaded] = useUserSettingsCompatibility(
+      COLUMN_MANAGEMENT_CONFIGMAP_KEY,
+      COLUMN_MANAGEMENT_LOCAL_STORAGE_KEY,
+      null,
```

Comment:
Please keep the default as undefined so that this value are not stored in user settings configmap until the user changes the table config.